docs/guide/custom-functions.md

```diff
@@ -140,16 +220,72 @@ the calculations.
 // arguments here are displayed just to show the structure
 public hyper(ast, state) {
     return 'Hyperformula'.length;
-  }
-};
+  };
+```
+
+### `runFunction()`
+
+To test your function's configuration, run the built-in `runFunction()` function.
+
+The `runFunction()` function:
+- Validates your function's [optional parameter](#optional-parameters) settings
+- Validates your [argument validation options](#argument-validation-options)
+- Checks if values returned by your function are in the right format
+
+```javascript
+import { FunctionPlugin } from 'hyperformula';
+
+export class CountHF extends FunctionPlugin {
+
+  public static implementedFunctions = {
+    'HYPER': {
+      method: 'hyper',
+      // set your optional parameters
+      arrayFunction: false,
+      // set your argument validation options
+      parameters: {
+        passSubtype: false
+      }
+    }
+  };
+
+  public hyper(ast, state) {
+    this.runFunction(HYPER())
+    
+    return 'Hyperformula'.length
+    }
+  };
+```
+
+### Throwing errors
+
+If you want your custom function to throw an error, check the [API reference](../api) for the HyperFormula error types.
+
+For example, if you want to throw a custom error message when the user doesn't pass enough arguments to your function, use the [`CellError` class](../api/classes/cellerror.md):
+
+```javascript
+// import `CellError` and `ErrorType`
+import { FunctionPlugin, CellError, ErrorType } from "hyperformula";
+
+public hyper({ args }) {
+    if (!args.length) {
+      // create a `CellError` instance with an `ErrorType` of `ERROR`, and your custom error message
+      const error = new CellError(ErrorType.ERROR, 'Not enough arguments!');
```

Comment:
Also IMO we can choose more iconic error type so devs will recognize we're referring to spreadsheets cell errors here

```suggestion
      const error = new CellError(ErrorType.DIV_BY_ZERO, 'Not enough arguments!');
```

And add a description that it will be displayed as `#DIV/0!` and will be correctly translated

Reply:
@wojciechczerniak - done: 304dfa0